testutil/keeper/tokenomics.go

```diff
@@ -609,6 +634,76 @@ func NewTokenomicsModuleKeepers(
 		Return(validators, nil).
 		AnyTimes()
 
+	// Set up delegation mocks if delegations are configured
+	if len(cfg.delegations) > 0 {
+		// Mock GetValidatorDelegations for each validator
+		for _, validator := range validators {
+			delegations, exists := cfg.delegations[validator.OperatorAddress]
+			if !exists {
+				delegations = []stakingtypes.Delegation{} // Empty slice for validators with no delegations
+			}
+
+			// Convert validator operator address to ValAddress for mock setup
+			valAddr, err := cosmostypes.ValAddressFromBech32(validator.OperatorAddress)
+			if err != nil {
+				panic(fmt.Sprintf("invalid validator operator address: %v", err))
+			}
+
+			mockStakingKeeper.EXPECT().
+				GetValidatorDelegations(gomock.Any(), valAddr).
+				Return(delegations, nil).
+				AnyTimes()
+		}
+
+		// Mock GetDelegatorDelegations for each delegator
+		for _, delegatorAddr := range cfg.delegatorAddresses {
+			// Find all delegations for this delegator across all validators
+			var delegatorDelegations []stakingtypes.Delegation
+			for _, validatorDelegations := range cfg.delegations {
+				for _, delegation := range validatorDelegations {
+					if delegation.DelegatorAddress == delegatorAddr.String() {
+						delegatorDelegations = append(delegatorDelegations, delegation)
+					}
+				}
+			}
+
+			mockStakingKeeper.EXPECT().
+				GetDelegatorDelegations(gomock.Any(), delegatorAddr, gomock.Any()).
+				Return(delegatorDelegations, nil).
+				AnyTimes()
+		}
+
+		// Mock Validator method for individual validator queries
+		for _, validator := range validators {
+			valAddr, err := cosmostypes.ValAddressFromBech32(validator.OperatorAddress)
+			if err != nil {
+				panic(fmt.Sprintf("invalid validator operator address: %v", err))
+			}
+
+			mockStakingKeeper.EXPECT().
+				Validator(gomock.Any(), valAddr).
+				Return(&validator, nil).
+				AnyTimes()
+		}
```

Comment:
I feel like we should do this even if there aren't any delegations.

Reply:
We do, we just return an empty `[]stakingtypes.Delegation{}` slice regardless of the validator address. I think it makes it slightly more reusable:

```go
	// Set up delegation mocks if delegations are configured
	if len(cfg.delegations) > 0 {
	...
	} else {
		// Default mock setup for when no delegations are configured
		// Mock empty delegations for any validator
		mockStakingKeeper.EXPECT().
			GetValidatorDelegations(gomock.Any(), gomock.Any()).
			Return([]stakingtypes.Delegation{}, nil).
			AnyTimes()

	}
```